ignition/templates/factory.py

```diff
@@ -32,26 +33,47 @@ def __init__(self, driver_types, app_name, version="1.0.0", port=None, module_na
         self.port = port
         if module_name == None:
             module_name = self.generate_module_name(self.app_name)
+        self.__validate_module_name(module_name)
         self.module_name = module_name
         self.description = description
         if docker_name == None:
             docker_name = self.generate_docker_name(self.app_name)
+        self.__validate_docker_name(docker_name)
         self.docker_name = docker_name
         if helm_name == None:
             helm_name = self.generate_helm_name(self.app_name)
+        self.__validate_helm_name(helm_name)
         self.helm_name = helm_name
         if helm_node_port == None:
             helm_node_port = self.generate_node_port()
         self.helm_node_port = helm_node_port
 
+    def __validate_module_name(self, module_name):
+        if not re.match('^[a-zA-Z0-9]*$', module_name):
+            raise ValueError('module_name must be a string with characters from a-z, A-Z, 0-9 but was: {0}'.format(module_name))
+
+    def __validate_helm_name(self, helm_name):
+        if not re.match('^[a-zA-Z0-9-_]*$', helm_name):
+            raise ValueError('helm_name must be a string with characters from a-z, A-Z, 0-9, dash (-) or underscore (_) but was: {0}'.format(helm_name))
+
+    def __validate_docker_name(self, docker_name):
+        if not re.match('^[a-zA-Z0-9-_]*$', docker_name):
+            raise ValueError('docker_name must be a string with characters from a-z, A-Z, 0-9, dash (-) or underscore (_) but was: {0}'.format(docker_name))
+    
     def generate_helm_name(self, app_name):
-        return app_name.lower().replace(" ", "-")
+        filtered_app_name = re.sub('[^A-Za-z0-9-_ ]+', '', app_name)
```

Comment:
minor: app_name -> helm_name?

Reply:
Well the app_name is the input, but I'll change the "filtered_app_name" to show it's no longer the app_name but the helm_name